lib/vector-core/src/event/proto.rs

```diff
@@ -675,11 +675,7 @@ impl From<Metadata> for EventMetadata {
             metadata = metadata.with_origin_metadata(origin_metadata.into());
         }
 
-        if let Ok(uuid) = Uuid::from_slice(&value.source_event_id) {
-            metadata = metadata.with_source_event_id(uuid);
-        } else {
-            error!("Invalid source_event_id in metadata");
-        }
+        metadata = metadata.with_source_event_id(Uuid::from_slice(&value.source_event_id).ok());
```

Comment:
It seems like this is ignoring errors. Do we want to still log a (rate limited) error if we receive an invalid id? We'd want to allow for empty string since that is what payloads from older versions of Vector will be deserializing as.

Reply:
Good point, I can re-add the log for that case